app/src/main/java/com/sdp13epfl2021/projmag/adapter/ProjectAdapter.kt

```diff
@@ -140,6 +141,23 @@ class ProjectAdapter(private val context: Context, private val utils: Utils, pri
         }
     }
 
+
+    private fun tagAndSectionsChipViewSetup(project: ImmutableProject, holder: ProjectViewHolder){
+        val green = ColorStateList.valueOf(ContextCompat.getColor(context, R.color.light_green))
+        val teal = ColorStateList.valueOf(ContextCompat.getColor(context, R.color.teal_700))
+        chipAdding(project.tags, holder, green)
+        chipAdding(project.allowedSections, holder,teal)
+    }
+
+    private fun chipAdding(list:List<Any>, holder:ProjectViewHolder, color:ColorStateList){
+        for(text in list){
+            val chipView: Chip = Chip(context)
+            chipView.text = text as CharSequence
```

Comment:
Unsafe cast can be dangerous in general.
```suggestion
    private fun chipAdding(list: List<String>, holder: ProjectViewHolder, color: ColorStateList) {
        for (text in list) {
            val chipView: Chip = Chip(context)
            chipView.text = text
```